indexing-service/src/main/java/io/druid/indexing/common/task/VersionConverterTask.java

```diff
@@ -142,7 +152,7 @@ public TaskStatus run(TaskToolbox toolbox) throws Exception
                 public Task apply(DataSegment segment)
                 {
                   final Integer segmentVersion = segment.getBinaryVersion();
-                  if (!CURR_VERSION_INTEGER.equals(segmentVersion)) {
+                  if (force || !CURR_VERSION_INTEGER.equals(segmentVersion)) {
```

Comment:
let's add a log message saying `Segment[%s] already at version[%s], forcing update` in case of forced update, similar to what we do below when we skip segments.